Dalamud/Interface/Textures/Internal/TextureManager.Drawer.cs

```diff
@@ -0,0 +1,398 @@
+using System.Buffers;
+using System.Diagnostics.CodeAnalysis;
+using System.Numerics;
+
+using Dalamud.Storage.Assets;
+using Dalamud.Utility;
+
+using ImGuiNET;
+
+using TerraFX.Interop.DirectX;
+using TerraFX.Interop.Windows;
+
+namespace Dalamud.Interface.Textures.Internal;
+
+/// <summary>Service responsible for loading and disposing ImGui texture wraps.</summary>
+internal sealed partial class TextureManager
+{
+    private SimpleDrawerImpl? simpleDrawer;
+
+    /// <summary>A class for drawing simple stuff.</summary>
+    [SuppressMessage(
+        "StyleCop.CSharp.LayoutRules",
+        "SA1519:Braces should not be omitted from multi-line child statement",
+        Justification = "Multiple fixed blocks")]
+    internal sealed unsafe class SimpleDrawerImpl : IDisposable
```

Comment:
If I understand this code and the usage correctly, this is only used for "copying" a texture by drawing it to another? I don't know a lot about DirectX, why is this necessary?

Reply:
`.Draw` should be taken as a hardware-accelerated pixel format conversion. `.StripAlpha` that follows will touch all pixels so that all alpha values become 1, which cannot be done with the functions exposed from ImGui, and would be noticeably more time consuming if the process is done by taking the pixels off GPU, process it with CPU, and upload to GPU again.